test_programs/execution_success/sha256_regression_5836/src/main.nr

```diff
@@ -0,0 +1,4 @@
+fn main(headers: [u8; 472], result: pub [u8; 32]) {
+    let hash = std::hash::sha256_var(headers, headers.len() as u64);
```

Comment:
Each of the other cases returned the correct results, but some more under-constrained warnings were found. Turns out I had to declare a new witness `zero` that was connected to the function inputs in order to avoid the under-constrained bug warnings when trying to assert against zero...